tests/upgrade/rules/test_hostname_callable_rule.py

```diff
@@ -0,0 +1,36 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+from unittest import TestCase
+
+from airflow.upgrade.rules.hostname_callable_rule import HostnameCallable
+from tests.test_utils.config import conf_vars
+
+
+class TestFernetEnabledRule(TestCase):
+    @conf_vars({("core", "hostname_callable"): "dummyhostname:function"})
+    def test_incorrect_hostname(self):
+        result = HostnameCallable().check()
+        self.assertEqual(
+            result,
+            "Error: hostname_callable dummyhostname:function "
+            "contains a colon instead of a dot. please change to dummyhostname.function",
+        )
+
+    @conf_vars({("core", "hostname_callable"): "dummyhostname.function"})
```

Comment:
Is this the correct value for Airflow 1.10? I imagine that we are able to ensure compatibility between the new and the old value at the same time. This will facilitate migrations as users will not have to change everything in one step, but they will be able to fix and test each value step by step.

Reply:
@mik-laj oof good catch. Should I add the backwards compat in this PR?

Reply:
Personally, I don't think it's necessary to make this change to Airflow 1.10. When the user updates Airflow to version 2.0, everything still works, so we don't have to worry too much to make this change in Airflow 1.10. It is critical to discover all the breaking changes in airflow upgrade-check.  All changes that are not breaking are nice-to-have.

Reply:
2.0 will read the 1.10 style config value and upgrade-in-memory+warn. https://github.com/apache/airflow/commit/ce50538503a4e670830ff26cfa42c162049e9253